### ISSUE TITLE
build on develop branch broken on windows platform

### ISSUE THREAD

**Issue by fkleedorfer** [eclipse/rdf4j#3473]
### Current Behavior

A recent change seems to have broken the build, tests are failing in `rdf4j-queryalgebra`. 

It seems some tests, such as the `org.eclipse.rdf4j.query.algebra.evaluation.impl.QueryCostEstimatesTest` expect the `toString()` value of a `TupleExpr` with the `\n` line separator, but the string is generated with `\r\n` line separator.

### Expected Behavior

These tests should not fail.

### Version

4.0.0.SNAPSHOT

### Are you interested in contributing a solution yourself?

No

**Comment by fkleedorfer** [comment]
Dug a little. Seems that these tests are platform-dependent: The `org.eclipse.rdf4j.query.algebra.helpers.QueryModelTreePrinter` uses 
```
private static final String LINE_SEPARATOR = System.getProperty("line.separator");
```
If github runs its builds on windows machines, that might explain the failure. But why has it worked until now? Hmm

A possible fix is to use the same, platform-dependent line separator in the test.

**Comment by fkleedorfer** [comment]
In the `org.eclipse.rdf4j.query.algebra.evaluation.impl.FilterOptimizerTest` it seems the expected `TupleExpr` should be wrapped in a `QueryRoot` element, or alternatively, the QueryRoot should not be in the actual result.

**Comment by fkleedorfer** [comment]
The `org.eclipse.rdf4j.sail.memory.QueryPlanRetrievalTest` also has failing tests

**Comment by fkleedorfer** [comment]
Fixed in #3474
